Analysis/include/BaseEventAnalyzer.h

```diff
@@ -65,6 +65,7 @@ class BaseEventAnalyzer : public EventAnalyzerCore {
     std::shared_ptr<NonResModel> nonResModel;
     const std::vector<std::string> trigger_patterns;
     std::shared_ptr<mc_corrections::EventWeights_HH> eventWeights_HH;
+    bbtautau::AnaTupleWriter::BTagWeights btag_weights;
```

Comment:
btag_weights change from one event to another => do not define it as a class member.